lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -907,8 +908,10 @@ def install_local(
     if dangerous:
         _cmd.append("--dangerous")
     try:
-        result = subprocess.check_output(_cmd, universal_newlines=True).splitlines()[0]
+        ansi_filter = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
```

Comment:
How did you pick that regex? If I scan through https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python I find tons of answers including 

```
ansi_regex = r'\x1b(' \
             r'(\[\??\d+[hl])|' \
             r'([=<>a-kzNM78])|' \
             r'([\(\)][a-b0-2])|' \
             r'(\[\d{0,2}[ma-dgkjqi])|' \
             r'(\[\d+;\d+[hfy]?)|' \
             r'(\[;?[hf])|' \
             r'(#[3-68])|' \
             r'([01356]n)|' \
             r'(O[mlnp-z]?)|' \
             r'(/Z)|' \
             r'(\d+)|' \
             r'(\[\?\d;\d0c)|' \
             r'(\d;\dR))'
```

which looks very different.
Maybe add a comment explaining your choice?
And perhaps a couple of tests :)

Reply:
There are quite a few possible expressions for filtering out ANSI escape sequences. The expression that I am using only filters out 7-bit C1 ANSI sequences. If I found the answer that contains the regular expression copied above, the expression does not actually work. It incompletely filters out the escape sequences; it does not remove certain parts of the ANSI escape sequences from the string.

I'll add a comment that says the regular expression removes 7-bit C1 ANSI escape sequences, and I'll see about adding some tests that filter the ANSI sequences out of captured output from `snap install ...`.

Reply:
Added tests in https://github.com/canonical/operator-libs-linux/pull/56/commits/39ab84cc5baf45aef7a799849cdefe61ef181b31 and I added a comment explaining what the regular expression does in https://github.com/canonical/operator-libs-linux/pull/56/commits/8b13e63cdd98108dfd70e978ad4db8a225985b94. I needed to push a extra commit however because the linter was not happy with the formatting of the test I added.